nav2_costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -213,47 +203,80 @@ void ObstacleLayer::onInitialize()
       observation_notifiers_.back()->setTargetFrames(target_frames);
     }
   }
+   setupDynamicReconfigure(node_);
+} */
+    // TODO(bpwilcox): replace with message filters above when enabled
+    // create a callback for the topic
+    auto buffer = observation_buffers_.back();
+    if (data_type == "LaserScan") {
```

Comment:
There's a lot of redundancy in this code block. I realize it is temporary code and should go away when message filters is enabled, but who knows how long temporary is. I'll leave it up to you whether you want to fix it or leave it alone such that your name comes up when somebody runs `git blame` ;-)

Reply:
The message filters code is equally redundant, my format is pretty much the same. I'm ok leaving as is for now.

Reply:
Ok. For future reference, I think what I'd have done here is just define a macro at the start of the function and then used that to simplify the subscription creation code.

```
#define CREATE_SUBSCRIBER(message_type, callback) \
  node_->create_subscription<message_type>(topic, [&, buffer] (message_type::ConstSharedPtr message) \ 
          { callback(message, buffer); }, custom_qos_profile);

...
if (data_type == "LaserScan") {
  if (inf_is_valid) {
   sub = CREATE_SUBSCRIBER(sensor_msgs::msg::LaserScan, laserScanValidInfCallback);
  } else {
    sub = CREATE_SUBSCRIBER(sensor_msgs::msg::LaserScan, laserScanValidCallback);
  }
} else {
  if (inf_is_valid) {
   RCLCPP_WARN ...
  } else {
    sub = CREATE_SUBSCRIBER(sensor_msgs::msg::PointCloud2, pointCloud2Callback);
  }
}
```